lm_eval/models/gpt2.py

```diff
@@ -243,3 +151,83 @@ def _collate(x):
             res.append(s)
         
         return reord.get_original(res)
+
+
+class HFLM(TorchLM):
+
+    def __init__(self, device='cuda', pretrained='gpt2', revision='main', subfolder=None, tokenizer=None, batch_size=1):
+        super().__init__()
+
+        assert isinstance(device, str)
+        assert isinstance(pretrained, str)
+        assert isinstance(batch_size, int)
+
+        if device:
+            self.device = torch.device(device)
+        else:
+            self.device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+
+        # TODO: update this to be less of a hack once subfolder is fixed in HF
+        self.gpt2 = transformers.AutoModelForCausalLM.from_pretrained(pretrained, revision=revision +("/" + subfolder if subfolder is not None else "")).to(self.device)
```

Comment:
Maybe `self.gpt2` should be renamed to a more generic term following the widened scope of this file?